src/OSPSuite.Core/Domain/Services/ResidualCalculator.cs

```diff
@@ -99,20 +99,45 @@ private IReadOnlyList<Residual> calculateOutputResiduals(DataColumn simulationCo
          {
             var weight = outputMapping.Weight * outputMapping.WeightedObservedData.Weights[index];
             var observedValue = convertToBaseUnit(mergedDimension, currentObservedDataUnit, observedValueColumn[index]);
-            if (!observedValue.IsValid())
-               continue;
 
             var observedTime = observedTimeColumn[index];
-            var simulatedValue = simulationColumn.GetValue(observedTime);
-            if (!simulatedValue.IsValid())
+
+            if(simulationTimeIsTooShort(simulationColumn, observedTime))
+               continue;
```

Comment:
Small optimization: this check could pe put at the very beginning of the for-loop.
(in order to avoid the calculations in the lines 100-101 in case sim time is too short).